examples/docker/basic_usage.py

```diff
@@ -0,0 +1,19 @@
+from pipeline import docker
+from pipeline import Pipeline, Variable, pipeline_function
+
+
+@pipeline_function
+def add_numbers(input_dict: dict) -> float:
+    return input_dict["in_1"] + input_dict["in_2"]
+
+
+with Pipeline("AddNumbers") as builder:
+    in_1 = Variable(dict, is_input=True)
+    builder.add_variables(in_1)
+
+    sum = add_numbers(in_1)
+
+    builder.output(sum)
+
+
+docker.create_pipeline_api([Pipeline.get_pipeline("AddNumbers")])
```

Comment:
could we perhaps add a method under `Pipeline` abstracting this to be something like `Pipeline.dockerize(id: str)`

where it can invoke the functions from the `docker` module

Reply:
Happy to add another function for it, but still would like to keep this.
Personally i'm not a fan of doing it through `Pipeline` especially later when we start pulling pipelines from the cloud and then will have to probably use them as objects as well.

Reply:
That is what I meant, add just an abstraction of this onto the pipeline, a sugarcoat.

when we pull them from the cloud, wont we pull dags(the graphs)? I had the understanding that `Pipeline` is a manager os all of the dags

Reply:
it could be even written as `Pipeline.dockerize(ids: List[str])` and check for the graphs under the hood

Reply:
Gotcha, I must have missed that, we'll need to revert that my bad for not being clear

Reply:
no problem, we can also rename them, make local the default

Reply:
> Happy to add another function for it, but still would like to keep this. Personally i'm not a fan of doing it through `Pipeline` especially later when we start pulling pipelines from the cloud and then will have to probably use them as objects as well.

this applies to the graphs -- which in the schemas are modeled under the name pipeline, but on our library logic the object called Pipeline is a `bag of dags` a high level manager of pipelines actually -- perhaps we should consider naming convention, but with the code as is, I think the best way to keep things clean is to manage pipelines via this high level manager object called `Pipeline`

Reply:
I was pretty confused by the `Pipeline`/`Graph` difference, as to me 'Pipeline' is the graph whereas in this package the name for that is `Graph`. I would be in favour of dropping `Graph` and having everything under `Pipeline`, as things like `Pipeline.retrieve_remote` can be thought of as a 'constructor' of a `Pipeline`.